lib/ukstore/include/uk/store.h

```diff
@@ -29,3 +29,345 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+
+#ifndef __UK_STORE_H__
+#define __UK_STORE_H__
+
+#include <string.h>
```

Comment:
When the library is not part of the built, this header that is always available should not introduce unnecessary dependencies. Your library can't express any if not part of a build.

Reply:
So I would move header inclusion to line 97.

Reply:
When moving, make sure that headers that are required for definitions wheen the library is off, stay here. I hope these are only general unikraft headers (`/include`) and/or libc provided headers.

Reply:
There are no library dependencies when the library is off, it's fine